### PR TITLE
add Target.criticalSectionSize

### PR DESCRIPTION
**Pull request by WalterBright**
Eliminates a lot of mutable global accesses.

### CODE REVIEW

File: src/dmd/target.h [hunk 1]
```diff
@@ -37,6 +37,7 @@ struct Target
     // C ABI
     unsigned c_longsize;         // size of a C 'long' or 'unsigned long' type
     unsigned c_long_doublesize;  // size of a C 'long double'
+    uint criticalSectionSize;    // size of os critical section
```
Comment (thewilsonator) [inline src/dmd/target.h]: `unsigned`

**Review by thewilsonator — COMMENTED** [review]
Otherwise LGTM.

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/target.h (hunk 1 above)
Comment (WalterBright) [reply src/dmd/target.h]: I should have simply made `unsigned` work in D.

**Comment by ibuclaw** [comment]
Only that it would be nice to move this to druntime.

Does the current mechanism even suitably align the temp buffer? I think no, so use of synchronize may crash on platforms where it is strictly required for pthread_mutex_t to be on an 8-byte boundary (for example, Solaris).

**Pull request merged**